src/store/storeConfig.ts

```diff
@@ -255,6 +255,7 @@ export const storeConfig: StoreOptions<State> = {
           // Updates the selected mesh if one wasn’t read earlier or if it’s not an existing mesh.
           const newStoredMesh = ClientStorage.get('selectedMesh')
           if (newStoredMesh === null || !state.meshes.items.some((mesh) => mesh.name === newStoredMesh)) {
+            await dispatch('sidebar/getInsights')
```

Comment:
From what I remember from last time we said that we should be validating the value of `selectedMesh` before we use it for `sidebar/getInsights`, is this to address that, and if so do we need to remove line 249 (https://github.com/kumahq/kuma-gui/pull/583/files#diff-7487c8e94db54374c1bf362cbee3cce6a715793c1b2fba31289e684994fd154cR249) where we call `sidebar/getInsights`, otherwise this request gets made twice does it?

I also noticed when trying this out that when I have `selectedMesh:all` in localStorage it still makes a HTTP request asking for meshInsights for `all`. Everything seems to fail nicely still, but we shouldn't be making that request at all ideally. I think this code here is possibly all related to that issue.

Super fine by me to address that in a future PR if you didn't want to do it here. This PR is an improvement on its own. Lemme know what you want to do.

Reply:
I’ve decided to rewrite the logic of this action to be more minimal.

Essentially, the call to `dispatch('sidebar/getInsights')` is now delayed as long as possible. Before doing that, we now check in order whether a stored mesh is present and whether it exists. If meshes are present, we fall back to the first one if the stored mesh can’t be found. If meshes aren’t present at all, we clear the stored value.